### PR TITLE
feat: add disk encryption settings

### PR DESCRIPTION
**Pull request by sefornes**
- Add variables for `disk_encryption_set_id` for both OS Disk and Data Disks

### CODE REVIEW

**Comment by sefornes** [comment]
Output from `terraform test`:
```
tests\unit.tftest.hcl... in progress
  run "setup_tests"... pass
  run "linux_vm"... pass
  run "windows_vm"... pass
  run "network_interface_nsg_association"... pass
  run "enable_system_assigned_identity"... pass
  run "disable_system_assigned_identity"... pass
  run "user_assigned_identities"... pass
  run "custom_data"... pass
  run "data_disks"... pass
tests\unit.tftest.hcl... tearing down
tests\unit.tftest.hcl... pass
```

**Comment by sefornes** [comment]
Output from `terraform test`:
```
tests\unit.tftest.hcl... in progress
  run "setup_tests"... pass
  run "linux_vm"... pass
  run "windows_vm"... pass
  run "network_interface_nsg_association"... pass
  run "enable_system_assigned_identity"... pass
  run "disable_system_assigned_identity"... pass
  run "user_assigned_identities"... pass
  run "custom_data"... pass
  run "data_disks"... pass
tests\unit.tftest.hcl... tearing down
tests\unit.tftest.hcl... pass
```

File: variables.tf [hunk 1]
```diff
@@ -243,3 +243,17 @@ variable "license_type" {
...
+  default     = null
+  nullable    = true
+}
+
+variable "data_disk_encryption_set_id" {
+  description = "(Optional) The ID of the Disk Encryption Set which should be used to Encrypt data Disks."
+  type        = string
+  default     = null
+  nullable    = true
+}
```
Comment (hknutsen) [inline variables.tf]: I think this should be configurable on a per-disk-basis, i.e. it this standalone variable should be removed and an property `disk_encryption_set_it` of type `optional(string)` should be added to this existing variable instead:

https://github.com/equinor/terraform-azurerm-vm/blob/7ba88873c71d2223e32201997792a196be13c9c3/variables.tf#L190-L202

File: variables.tf [hunk 2]
```diff
@@ -243,3 +243,17 @@ variable "license_type" {
     error_message = "License type must be \"None\", \"Windows_Client\", \"Windows_Server\", \"RHEL_BYOS\" or \"SLES_BYOS\"."
   }
 }
+
+variable "os_disk_encryption_set_id" {
+  description = "(Optional) The ID of the Disk Encryption Set which should be used to Encrypt this OS Disk."
```
Comment (hknutsen) [inline variables.tf]: Terraform Registry automatically marks variables as optional:

```suggestion
  description = "The ID of the Disk Encryption Set which should be used to encrypt this OS disk."
```

**Review by hknutsen — CHANGES_REQUESTED** [review]

File: variables.tf (hunk 1 above)
Comment (sefornes) [reply variables.tf]: Good input. Implemented.

**Review by hknutsen — COMMENTED** [review]
Perfect 🤩

**Review by hknutsen — APPROVED** [review]

**Pull request merged**